### PR TITLE
Changes to SDF API to use DoFn Params

### PR DESCRIPTION
**Pull request by pabloem**
A minimal change to the SDF API for consistency and clarity of expectations.

### CODE REVIEW

File: sdks/python/apache_beam/runners/common.py [hunk 1]
```diff
@@ -239,8 +239,8 @@ def __init__(self, do_fn):
 
   def get_restriction_provider(self):
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/common.py]: Rename the method as well?

Comment (aaltay) [reply sdks/python/apache_beam/runners/common.py]: Scratch that.

**Comment by charlesccychen** [comment]
I think you may need to update the invoker logic in common.py: https://github.com/apache/beam/blob/ef67378fec2afadbe4a5303e61ac0427ced1faa0/sdks/python/apache_beam/runners/common.py#L477

**Comment by pabloem** [comment]
Thanks for looking @charlesccychen. I believe the parameter is repalced in invoke_process, and the relevant change is in line 541: https://github.com/apache/beam/pull/8430/files#diff-dc4f89b2c45e945478f3fb5a0c482976R541

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
In Java, users provide a `TableDestination` object. The `TableDestination` interface has individual functions for each table attribute, and doesn't currently support clustering - only time partitioning. The API that we are defining allows us to support time partitioning, and clustering (and any new BQ feature if necessary).

See https://beam.apache.org/releases/javadoc/2.12.0/org/apache/beam/sdk/io/gcp/bigquery/TableDestination.html

**Comment by pabloem** [comment]
Run Python PostCommit

**Comment by pabloem** [comment]
r: @robertwb  PTAL : )

**Comment by pabloem** [comment]
This may require updates to documentation elsewhere.

**Pull request merged**